### ISSUE TITLE
RpcTestWithSpawn.test_stress_heavy_rpc is flaky in CI

### ISSUE THREAD

**Issue by mruberry** [pytorch/pytorch#29410]
Build: pytorch_macos_10_13_py3_test

See: https://app.circleci.com/jobs/github/pytorch/pytorch/3514074

```
Nov 07 12:26:48 ERROR: test_stress_heavy_rpc (__main__.RpcTestWithSpawn)
Nov 07 12:26:48 ----------------------------------------------------------------------
Nov 07 12:26:48 Traceback (most recent call last):
Nov 07 12:26:48   File "/Users/distiller/project/test/common_distributed.py", line 130, in wrapper
Nov 07 12:26:48     self._join_processes(fn)
Nov 07 12:26:48   File "/Users/distiller/project/test/common_distributed.py", line 211, in _join_processes
Nov 07 12:26:48     self._check_return_codes(elapsed_time)
Nov 07 12:26:48   File "/Users/distiller/project/test/common_distributed.py", line 230, in _check_return_codes
Nov 07 12:26:48     raise RuntimeError('Process {} terminated or timed out after {} seconds'.format(i, elapsed_time))
Nov 07 12:26:48 RuntimeError: Process 0 terminated or timed out after 300.0058991909027 seconds
Nov 07 12:26:48 
Nov 07 12:26:48 ----------------------------------------------------------------------
Nov 07 12:26:48 Ran 53 tests in 349.255s
```

@mrshenli @aazzolini

cc @ezyang @gchanan @zou3519 @jerryzh168 @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @rohan-varma @xush6528

**Comment by mrshenli** [comment]
This will be addressed by #29069 

```
Nov 07 12:21:47 test_stress_heavy_rpc (__main__.RpcTestWithSpawn) ... libc++abi.dylib: terminating with uncaught exception of type c10::Error: Need to initialize distributed autograd using torch.distributed.autograd.init() (getInstance at ../torch/csrc/distributed/autograd/context/dist_autograd_container.cpp:57)
Nov 07 12:21:47 frame #0: c10::Error::Error(c10::SourceLocation, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) + 135 (0x10ff41557 in libc10.dylib)
Nov 07 12:21:47 frame #1: torch::distributed::autograd::DistAutogradContainer::getInstance() + 500 (0x11d2adfc4 in libtorch.dylib)
Nov 07 12:21:47 frame #2: torch::distributed::rpc::RequestCallback::operator()(torch::distributed::rpc::Message&) const + 37 (0x11d2c8935 in libtorch.dylib)
Nov 07 12:21:47 frame #3: std::__1::__function::__func<std::__1::__bind<torch::distributed::rpc::ProcessGroupAgent::enqueueRecv(torch::distributed::rpc::RecvWork)::$_2, torch::distributed::rpc::RecvWork>, std::__1::allocator<std::__1::__bind<torch::distributed::rpc::ProcessGroupAgent::enqueueRecv(torch::distributed::rpc::RecvWork)::$_2, torch::distributed::rpc::RecvWork> >, void ()>::operator()() + 1133 (0x1199c816d in libtorch_python.dylib)
Nov 07 12:21:47 frame #4: c10::ThreadPool::main_loop(unsigned long) + 527 (0x10ff3c90f in libc10.dylib)
Nov 07 12:21:47 frame #5: std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, c10::ThreadPool::ThreadPool(int, int, std::__1::function<void ()>)::$_0> >(void*, void*) + 67 (0x10ff3ce23 in libc10.dylib)
Nov 07 12:21:47 frame #6: _pthread_body + 180 (0x7fffe929393b in libsystem_pthread.dylib)
Nov 07 12:21:47 frame #7: _pthread_body + 0 (0x7fffe9293887 in libsystem_pthread.dylib)
Nov 07 12:21:47 frame #8: thread_start + 13 (0x7fffe929308d in libsystem_pthread.dylib)
```

**Comment by mrshenli** [comment]
closed by #29069
